frontend/src/components/ViewActivities/index.js

```diff
@@ -8,12 +15,71 @@ import ActivityList from './activitylist.js';
  * - `tripId` {string} The trip's ID. This is sent to the component through the URL. 
  */
 class ViewActivities extends React.Component {
+  constructor(props) {
```

Comment:
Heads up that one of my PRs also edits this file substantially, so there will definitely be conflicts depending on which branch gets merged second. Feel free to reach out if this one is merged after, and if mine is merged after I'll very likely reach out to you so we can sync our code.